File: internal/domain/transaction.go
package domain

import "time"

type Transaction struct {
	ID              string    `json:"id"`
	BankID          string    `json:"bank_id"`
	Type            string    `json:"type"`
	Amount          float64   `json:"amount"`
	Currency        string    `json:"currency"`
	Description     string    `json:"description"`
	TransactionDate time.Time `json:"transaction_date"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

File: backend/src/types/Transaction.ts
import { Field, ObjectType, ID } from 'type-graphql'
import { Decimal } from '@prisma/client/runtime/library'

@ObjectType()
export class TransactionType {
  @Field(() => ID)
  id: string

  @Field(() => Date)
  createdAt: Date

  @Field(() => Date)
  updatedAt: Date

  @Field(() => String)
  bankId: string

  @Field(() => String, { nullable: true })
  type?: string | null

  @Field(() => Number, { nullable: true })
  amount?: Decimal | null

  @Field(() => String, { nullable: true })
  currency?: string | null

  @Field(() => String, { nullable: true })
  description?: string | null

  @Field(() => Date, { nullable: true })
  transactionDate?: Date | null
}
